lib/index.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+
+exports.register = (server, options, next) => {
+
+  server.ext('onPreResponse', (request, reply) => {
+
+    const settings = request.route.settings.plugins.totalCount;
+    const credentials = request.auth.credentials;
+    const query = request.query;
+    const filter = request.query.filter || {};
+
+    if (settings && credentials && query.include && query.include.indexOf('total_count') !== -1 && request.response.source) {
+
+      Bluebird.resolve(settings.model)
+      .then((Model) => {
+        if (typeof new Model().filter === 'function') {
```

Comment:
nit: does checking `Model.prototype.filter` work instead? might be a bit cleaner than instantiating a new object.